jni/src/nmslib_wrapper.cpp

```diff
@@ -243,14 +247,24 @@ jobjectArray knn_jni::nmslib_wrapper::QueryIndex(knn_jni::JNIUtilInterface * jni
         jniUtil->ReleaseFloatArrayElements(env, queryVectorJ, rawQueryvector, JNI_ABORT);
         throw;
     }
+
     jniUtil->ReleaseFloatArrayElements(env, queryVectorJ, rawQueryvector, JNI_ABORT);
+    std::unordered_map<std::string, jobject> methodParams;
+    if (methodParamsJ != nullptr) {
+        methodParams = jniUtil->ConvertJavaMapToCppMap(env, methodParamsJ);
+    }
 
-    similarity::KNNQuery<float> knnQuery(*(indexWrapper->space), queryObject.get(), kJ);
-    indexWrapper->index->Search(&knnQuery);
+    int queryEfSearch = knn_jni::commons::getIntegerMethodParameter(env, jniUtil, methodParams, EF_SEARCH, -1);
+    similarity::KNNQuery<float>* query;
```

Comment:
Can we use smart pointer to avoid having to delete at end?

Reply:
+1

Reply:
As far as I remember, I had a really hard time getting the unit test right with a smart pointer so had to switch to this. Suggestions are welcome here. I would like to keep it as it is otherwise

Reply:
Can we create a GH issue to refactor the code so that we can write unit test with smart pointer?

Reply:
https://github.com/opensearch-project/k-NN/issues/1785